main.go

```diff
@@ -91,29 +93,25 @@ func getEtcdConfig() (etcd.Config, error) {
 	return config, nil
 }
 
-func getGlobalASN(api etcd.KeysAPI) (uint32, error) {
-	res, err := api.Get(context.Background(), fmt.Sprintf("%s/global/as_num", CALICO_BGP), nil)
-	if err != nil {
-		return 0, err
-	}
-	asn, err := strconv.ParseUint(res.Node.Value, 10, 32)
-	if err != nil {
-		return 0, err
+func getGlobalASN(api etcd.KeysAPI) (numorstring.ASNumber, error) {
+	value := os.Getenv(AS)
+	if value == "" {
+		res, err := api.Get(context.Background(), fmt.Sprintf("%s/global/as_num", CALICO_BGP), nil)
+		if err != nil {
+			return 0, err
+		}
+		value = res.Node.Value
 	}
-	return uint32(asn), nil
+	return numorstring.ASNumberFromString(value)
 }
 
-func getPeerASN(api etcd.KeysAPI, host string) (uint32, error) {
+func getPeerASN(api etcd.KeysAPI, host string) (numorstring.ASNumber, error) {
```

Comment:
you should be able to use libcalico `Node` resource here to get the host ASN.
see: https://github.com/projectcalico/libcalico-go/blob/master/lib/api/node.go#L62